### ISSUE TITLE
labelField inside Collection doesn't show the expected values

### ISSUE THREAD

**Issue by PKuhlmay** [FriendsOfTYPO3/content-blocks#112]
I have the following Collection:

```
  - identifier: podcasts
    type: Collection
    labelField:
      - podcast_episode
      - podcast_date
      - podcast_title
    fields:
      - identifier: podcast_header_settings
        type: Palette
        fields:
        - identifier: podcast_episode
          type: Number
          range:
            lower: 1
        - identifier: podcast_date
          type: DateTime
          format: date
          disableAgeDisplay: false
      - identifier: podcast_title
        type: Text
      - identifier: podcast_description
        type: Textarea
        enableRichtext: true
      - identifier: podcast_link
        type: Link
        allowedTypes:
          - page
```

The problem is, that the configuration for labelField doesn't work as expected. I expected to see the podcast_episode (which is a number), the date and the title. But I get the title, the date and the title. So the title is shown twice.

![Bildschirmfoto 2024-01-31 um 17 34 27](https://github.com/nhovratov/content-blocks/assets/107543296/9447f81f-8ce7-4c92-8feb-c9523fd6ba82)

**Comment by nhovratov** [comment]
Good catch. Yes, this is a bug. So, right now only Text, Textarea and Email are allowed as label field. Because episode is a Number, Content Blocks searches for the first valid field (which is podcast_title) and uses it as the primary field. The other fields are not checked at all!

To fix this issue we have to think about:
- What are valid field types for the labelField ?
- What to do, if the user defined invalid types? Force their usage, or throw them away (could lead to unexpected behaviour).
- Should it cause an exception if a wrong field type is chosen?

I'm happy to discuss these questions here or in the next call on monday.

**Comment by PKuhlmay** [comment]
In my case it would be nice to have numbers as valid fields. Because this is a list of podcasts and the list would be clearer, if the episode number comes at first, instead of only the title.

**Comment by PKuhlmay** [comment]
It is getting a little bit weird.

I have the same configuration like above.
When I set the labelFields to
```
  - date
  - episode
  - date
  - title
```

The episode gets displayed now. Only the first "date" is like a fallback to the title. It doesn't matter what I set at first position, when I set the episode as second position, it gets displayed.
![Bildschirmfoto 2024-02-01 um 11 25 49](https://github.com/nhovratov/content-blocks/assets/107543296/93d60147-e631-4b89-8027-4da73e3ae6d0)
![Bildschirmfoto 2024-02-01 um 11 25 39](https://github.com/nhovratov/content-blocks/assets/107543296/6a756d00-bd07-4148-85e9-5f1872b6c102)

And even with `disableAgeDisplay: false` for the type DateTime, I get this displayed inside my labelField.
